tools/pipelines/templates/include-install-pnpm.yml

```diff
@@ -48,5 +48,36 @@ steps:
       echo "Using node $(node --version)"
       sudo corepack enable
       echo "Using pnpm $(pnpm -v)"
+      # All below settings are user-level. We set the user-level .npmrc to be located in a well-known temp location
+      # that will persist for the duration of this job.
+      echo "##vso[task.setvariable variable=NPM_CONFIG_USERCONFIG]$(Agent.TempDirectory)/.npmrc"
+      echo "Pnpm user config location: $(pnpm config get userconfig)"
       pnpm config set store-dir ${{ parameters.pnpmStorePath }}
+      echo "Pnpm store: ${{ parameters.pnpmStorePath }}"
+      echo "Primary registry: ${NPM_REGISTRY}"
       pnpm config set -g workspace-concurrency 0
+      pnpm config set registry "${NPM_REGISTRY}"
+      if [ ${NPM_REGISTRY} == "https://registry.npmjs.org/" ]; then
+        echo "##vso[task.setvariable variable=registryType]public"
+      else
+        echo "##vso[task.setvariable variable=registryType]private"
+      fi
+  env:
+    NPM_REGISTRY: $(ado-feeds-primary-registry)
+    # The setvariable task in this script only sets the environment variable for subsequent tasks in this job.
+    # We use the same value here to ensure that all configuration applies throughout pnpm usage in this job.
+    NPM_CONFIG_USERCONFIG: $(Agent.TempDirectory)/.npmrc
+    # We should leverage the primary npm registry to install pnpm as well. However, ADO artifacts feeds do not implement
+    # the full npm registry API including a route that corepack uses to get package metadata--the version route here:
+    # https://github.com/nodejs/corepack/blob/bc13d40037d0b1bfd386e260ae741f55505b5c7c/sources/npmRegistryUtils.ts#L32
+    # Thus installing pnpm from an ADO feed using corepack is not possible at time of writing.
+    # COREPACK_NPM_REGISTRY: $(ado-feeds-primary-registry)
+
+# Authenticate to npm feed if required
+- task: npmAuthenticate@0
+  displayName: 'Npm authenticate'
+  condition: and(succeeded(), eq(variables['registryType'], 'private'))
+  retryCountOnTaskFailure: 1
+  inputs:
+    # All pnpm configuration above applies to the user-level .npmrc, which is explicitly positioned via NPM_CONFIG_USERCONFIG above.
+    workingFile: $(Agent.TempDirectory)/.npmrc
```

Comment:
I'd say let's make this value an output of the Bash script task so we don't have the path in two places, or a template parameter with a default and docs saying "you don't really need to change this".

Reply:
Yeah, good call. I went with a parameter as that seems simpler